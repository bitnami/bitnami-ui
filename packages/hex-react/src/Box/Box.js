import React from 'react';
import PropTypes from 'prop-types';
import cs from 'classnames';
import { Spacing, Background, Shadow, cleanProps } from '../helpers';

// Outside the Box!
const Box = props => {
  const { className, children } = props;

  // Setup the proper CSS classes and clean the props
  const helpers = [Spacing, Background, Shadow];
  const helperClasses = cs(className, ...helpers.map(h => h.propToClass(props)));
  const remainingProps = cleanProps(props, ...helpers);

  return (
    <div className={helperClasses} {...remainingProps}>
      {children}
    </div>
  );
};

// Values for spacing
const spacingValues = [
  'reset',
  'tiny',
  'small',
  'normal',
  'big',
  'bigger',
  'enormous',
  'giant',
  'huge',
];

Box.propTypes = {
  /**
   * Any element that will be wrapped by the Box
   */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  /**
   * Setup the margin-left and right as auto
   */
  marginCenter: PropTypes.bool,
  /**
   * Set the top margin
   */
  marginTop: PropTypes.oneOf(spacingValues),
  /**
   * Set the bottom margin
   */
  marginBottom: PropTypes.oneOf(spacingValues),
  /**
   * Set the left margin
   */
  marginLeft: PropTypes.oneOf(spacingValues),
  /**
   * Set the right margin
   */
  marginRight: PropTypes.oneOf(spacingValues),
  /**
   * Set the vertical margin (top and bottom)
   */
  marginVertical: PropTypes.oneOf(spacingValues),
  /**
   * Set the horizontal margin (left and right)
   */
  marginHorizontal: PropTypes.oneOf(spacingValues),
  /**
   * Set the top padding
   */
  paddingTop: PropTypes.oneOf(spacingValues),
  /**
   * Set the bottom padding
   */
  paddingBottom: PropTypes.oneOf(spacingValues),
  /**
   * Set the left padding
   */
  paddingLeft: PropTypes.oneOf(spacingValues),
  /**
   * Set the right padding
   */
  paddingRight: PropTypes.oneOf(spacingValues),
  /**
   * Set the vertical padding (top and bottom)
   */
  paddingVertical: PropTypes.oneOf(spacingValues),
  /**
   * Set the horizontal padding (left and right)
   */
  paddingHorizontal: PropTypes.oneOf(spacingValues),
  /**
   * Display a shadow. The level represents the visual elevation
   */
  shadow: PropTypes.oneOf([1, 2, 3, 4, 5]),
  /**
   * Setup a background. It's based on the styleguide colors
   */
  background: PropTypes.oneOf(['brand', 'accent', 'action', 'iron', 'dark', 'white']),
  /**
   * Setup a gradient background. It's based on the styleguide gradients
   */
  gradient: PropTypes.oneOf(['brand', 'accent', 'action', 'light', 'danger']),
  /**
   * Setup a different angle for the gradient
   */
  gradientAngle: PropTypes.oneOf([45, 90, 135, 180, 225, 270]),
};

Box.defaultProps = {
  children: undefined,
  marginCenter: undefined,
  marginTop: undefined,
  marginBottom: undefined,
  marginLeft: undefined,
  marginRight: undefined,
  marginVertical: undefined,
  marginHorizontal: undefined,
  paddingTop: undefined,
  paddingBottom: undefined,
  paddingLeft: undefined,
  paddingRight: undefined,
  paddingVertical: undefined,
  paddingHorizontal: undefined,
  shadow: undefined,
  background: undefined,
  gradient: undefined,
  gradientAngle: undefined,
};

export default Box;
