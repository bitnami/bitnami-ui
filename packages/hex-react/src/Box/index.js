import BoxWithHelpers from './Box';

export default BoxWithHelpers;
