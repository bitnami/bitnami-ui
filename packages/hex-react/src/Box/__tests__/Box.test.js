import React from 'react';
import { shallow } from 'enzyme';

import Box from '../Box';

describe('Box', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Box>Test</Box>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with Helpers', () => {
    const wrapper = shallow(
      <Box marginTop="small" background="brand" fontSize="big">
        Test
      </Box>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
