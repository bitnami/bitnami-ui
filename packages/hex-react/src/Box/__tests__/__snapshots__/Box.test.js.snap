// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box apply correctly the helpers 1`] = `
<div
  className="margin-t-small bg-brand"
>
  Test
</div>
`;

exports[`Box renders correctly 1`] = `
<div
  className=""
>
  Test
</div>
`;
